<!--
  Standard launch configuration for an app manager in standalone mode.
  i.e. no android, no multimaster concert. You should be able to play
  the basic apps (chirp and talker etc with this).
 -->
<launch>
  <!-- ****************************** Capabilities ****************************** -->
  <arg name="capabilities"                       default="false"/>                                              <!-- enable/disable a capability server -->
  <arg name="capabilities_parameters"            default="$(find rocon_app_manager)/param/capabilities.yaml" /> <!-- detailed parameter configuration for the providers -->
  <arg name="capabilities_blacklist"             default="[]" />                                                <!-- blacklist specific capabilities -->
  <arg name="capabilities_nodelet_manager_name"  default="capability_server_nodelet_manager" />
  <arg name="capabilities_package_whitelist"     default="[std_capabilities]" />                                <!-- get capabilities from these packages only -->
  <arg name="capabilities_server_name"           default="capability_server"/>

  <group if="$(arg capabilities)">
    <include file="$(find rocon_app_manager)/launch/includes/_capabilities.xml">
      <arg name="blacklist"            value="$(arg capabilities_blacklist)" />
      <arg name="nodelet_manager_name" value="$(arg capabilities_nodelet_manager_name)" />
      <arg name="name"                 value="$(arg capabilities_server_name)" />
      <arg name="package_whitelist"    value="$(arg capabilities_package_whitelist)" />
      <arg name="parameters"           value="$(arg capabilities_parameters)" />
    </include>
  </group>

  <!-- ****************************** Rapp Manager ****************************** -->
  <arg name="auto_rapp_installation" default="false"/>              <!-- http://wiki.ros.org/rocon_app_manager/Tutorials/indigo/Automatic Rapp Installation -->
  <arg name="auto_start_rapp"        default=""/>                   <!-- autostart a rapp, e.g. rocon_apps/talker -->
  <arg name="rapp_package_whitelist" default="[rocon_apps]"/>       <!-- comma separated list of package names -->
  <arg name="rapp_package_blacklist" default="[]"/>
  <arg name="robot_name"             default="Cybernetic Pirate"/>
  <arg name="robot_type"             default="pc"/>
  <arg name="robot_icon"             default="rocon_icons/cybernetic_pirate.png"/>
  <arg name="screen" default="true"/>  <!-- verbose output from running apps -->

  <include file="$(find rocon_app_manager)/launch/includes/_app_manager.xml">
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="robot_type" value="$(arg robot_type)" />
    <arg name="robot_icon" value="$(arg robot_icon)" />
    <arg name="rapp_package_whitelist" value="$(arg rapp_package_whitelist)" />
    <arg name="rapp_package_blacklist" value="$(arg rapp_package_blacklist)" />
    <arg name="auto_start_rapp" value="$(arg auto_start_rapp)" />
    <arg name="screen" value="$(arg screen)" />
    <arg name="auto_rapp_installation" value="$(arg auto_rapp_installation)" />
    <arg name="capability_server_name" default="$(arg capabilities_server_name)"/>
  </include>

  <!-- ***************************** Rocon Master Info ************************** -->
  <arg name="robot_description"      default="To err is human, to 'arr is pirate."/>

  <include file="$(find rocon_app_manager)/launch/includes/_rocon_master_info.xml">
    <arg name="name" value="$(arg robot_name)" />
    <arg name="description" value="$(arg robot_description)" />
    <arg name="icon" value="$(arg robot_icon)" />
  </include>

  <!-- ***************************** Interactions ******************************* -->
  <arg name="interactions"          default="false"/>
  <arg name="interactions_list"    default="[]"/>
  <arg name="rosbridge_address"    default="localhost"/>
  <arg name="rosbridge_port"       default="9090"/>

  <group if="$(arg interactions)">
    <include file="$(find rocon_app_manager)/launch/includes/_interactions.xml">
      <arg name="interactions_list" value="$(arg interactions_list)"/>
      <arg name="rosbridge_address" value="$(arg rosbridge_address)"/>
      <arg name="rosbridge_port"    value="$(arg rosbridge_port)"/>
    </include>
  </group>

  <!-- ******************************** Zeroconf ******************************** -->
  <arg name="zeroconf"      default="false"/>
  <arg name="zeroconf_name" default="$(arg robot_name)"/>
  <arg name="zeroconf_port" default="11311"/>

  <group if="$(arg zeroconf)">
    <include file="$(find rocon_app_manager)/launch/includes/_zeroconf.xml">
      <arg name="name" value="$(arg zeroconf_name)"/>
      <arg name="port" value="$(arg zeroconf_port)"/>
    </include>
  </group>
</launch>
